core/unit_test/TestTeam.hpp

```diff
@@ -290,6 +290,58 @@ class ReduceTeamFunctor {
   }
 };
 
+template <typename ScalarType, class DeviceType, class ScheduleType>
+class ArrayReduceTeamFunctor {
+ public:
+  using execution_space = DeviceType;
+  using policy_type     = Kokkos::TeamPolicy<ScheduleType, execution_space>;
+  using size_type       = typename execution_space::size_type;
+
+  using value_type      = ScalarType[];
+  size_type value_count = 3;
+
+  const size_type nwork;
+
+  KOKKOS_INLINE_FUNCTION
+  ArrayReduceTeamFunctor(const size_type &arg_nwork) : nwork(arg_nwork) {}
+
+  KOKKOS_INLINE_FUNCTION
+  ArrayReduceTeamFunctor(const ArrayReduceTeamFunctor &rhs)
+      : nwork(rhs.nwork) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void init(value_type dst) const {
+    dst[0] = 0;
+    dst[1] = 0;
+    dst[2] = 0;
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void join(value_type dst, const value_type src) const {
+    dst[0] += src[0];
+    dst[1] += src[1];
+    dst[2] += src[2];
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const typename policy_type::member_type ind,
```

Comment:
Not blocking but I am really not a fan of calling it "ind".
Also why are we not passing by ref?